lib/HLSL/LiveValueAnalysis.cpp

```diff
@@ -0,0 +1,1323 @@
+        int line = debugInfo->getLine();
+        std::string fullPath =
+          formatSourceLocation( m_rootPath, filePath, line );
+        Spacer << std::setfill( '=' ) << std::setw( fullPath.length() + 33 )
+          << std::right << "\n";
+        PrettyStr << Spacer.str();
+        VSStr << Spacer.str();
+        PrettyStr << "Live State for TraceRay call at " << fullPath << "\n";
+        VSStr << "Live State for TraceRay call at " << fullPath << "\n";
+        PrettyStr << Spacer.str();
+        VSStr << Spacer.str();
+      }
+      else {
+        break;
+      }
+    }
+    size_t regs = 0;
+    size_t detected = 0;
+    std::vector<Instruction*> DetectedInstr;
+    for( Instruction *I : m_spillsPerTraceCall[callSite] ) {
+      // Only count the values for which metadata exists.
+      // TODO: Report count of values for those which are not used metadata?
+      // this can lead to missing live values in some cases.
+      if( isa<ExtractValueInst>( I ) ) {
+        I = dyn_cast<Instruction>(I->getOperand( 0 ));
+      }
+      if( I->isUsedByMetadata() ) {
+        regs++;
+      }
+      else {
+        DetectedInstr.push_back( I );
+      }
+      detected++;
+    }
+    if( detected != regs )
+    {
+      PrettyStr << "** DEBUG ************\n";
+      VSStr << "** DEBUG ************\n";
+      PrettyStr << "Detected " << (int)detected << " live values but only " << (int)regs << " are used by metadata\n";
+      VSStr << "Detected " << (int)detected << " live values but only " << (int)regs << " are used by metadata\n";
+      for( auto I : DetectedInstr ) {
+        I->print( PrettyStr );
+        I->print( VSStr );
+        PrettyStr << "\n";
+        VSStr << "\n";
+      }
+      PrettyStr << "*********************\n";
+      VSStr << "*********************\n";
+    }
+    if( const llvm::DebugLoc &debugInfo = callSite->getDebugLoc() ) {
+      PrettyStr << "Total 32-bit registers: " << (int)regs << "\n\n";
+      VSStr << "Total 32-bit registers: " << (int)regs << "\n\n";
+    }
+
+    std::ostringstream TmpStr;
+    for( Instruction *I :
+      sortInstructions<Instruction *>( m_spillsPerTraceCall[callSite] ) ) {
+      TmpStr.str( "" );
+      {
+        if( isa<ExtractValueInst>( I ) ) {
+          I = dyn_cast<Instruction>(I->getOperand( 0 ));
+        }
+
+        if( I->isUsedByMetadata() ) {
+          unsigned int totalLineWidth = 0;
+
+          if( auto *L = LocalAsMetadata::getIfExists( I ) ) {
+            if( auto *MDV = MetadataAsValue::getIfExists( I->getContext(), L ) ) {
+              for( User *U : MDV->users() ) {
+                if( DbgValueInst *DVI = dyn_cast<DbgValueInst>(U) ) {
+
+                  TmpStr << translateValueToName( DVI, I );
+
+                  // Output USE locations
+                  std::vector<PHINode *> Phis;
+                  std::set<PHINode *> VisitedPhis;
+                  std::set<DILocation *> UseLocations;
+                  while( 1 ) {
```

Comment:
If you're going to use unconditional loops, I request they be consistent. ```while(1)``` or ```for(;;)```. If I remember correctly, the NVIDIA style guide advocates ```for(;;)```